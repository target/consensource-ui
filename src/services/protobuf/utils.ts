import {
  CertificateRegistryPayload,
  ICertificateRegistryPayload,
} from './compiled';

export interface PayloadInfo {
  payloadBytes: string | Buffer | NodeJS.TypedArray | DataView;
  inputs: string[];
  outputs: string[];
}

export const ACTIONS = CertificateRegistryPayload.Action;

/**
 * Encodes a CertificateRegistryPayload message
 * @param message CertificateRegistryPayload message or plain object to encode
 */
export function encodePayload(
  payload: ICertificateRegistryPayload,
): Uint8Array {
  return CertificateRegistryPayload.encode(payload).finish();
}
