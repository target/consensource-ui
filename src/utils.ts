// Util collection

/**
 * Selects keys from an object
 */
export const pluck = (obj: any, ...keys: Array<any>): any =>
    keys.reduce((o, k) => {
        if (typeof obj[k] !== 'undefined') {
            o[k] = obj[k];
        }
        return o;
    }, {});

/**
 * Merges objects, starting from left to right
 */
