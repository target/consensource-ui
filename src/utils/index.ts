export function getLocaleFromUnix(timestampSeconds: number) {
  const timestampMS = timestampSeconds * 1000;
  return new Date(timestampMS).toLocaleDateString();
}
