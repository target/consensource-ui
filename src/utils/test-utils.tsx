import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { StoresContext } from 'view/context';
import { stores } from 'stores';
import { theme } from 'config/theme';
import { ThemeProvider } from '@material-ui/core';
import CssBaseline from '@material-ui/core/CssBaseline';

const wrapper = ({ children }: any) => {
  return (
    <CssBaseline>
      <ThemeProvider theme={theme}>
        <StoresContext.Provider value={stores}>
          <MemoryRouter initialEntries={['/']}>{children}</MemoryRouter>
        </StoresContext.Provider>
      </ThemeProvider>
    </CssBaseline>
  );
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
