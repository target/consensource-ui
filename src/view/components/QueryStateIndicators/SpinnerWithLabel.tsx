import React, { FC } from 'react';
import {
  CircularProgress,
  Grid,
  Typography,
  TypographyProps,
  CircularProgressProps,
} from '@material-ui/core';

const SPINNER_SIZE = window.screen.width / 10;

export interface SpinnerWithLabelProps {
  spinnerSize?: CircularProgressProps['size'];
  typeVariant?: TypographyProps['variant'];
}

/**
 * `<CircularProgress />` component that displays with a label
 * beneath the spinner.
 *
 * Note that this component should primarily be used for loading
 * animations on top-level pages, components, etc. If there is
 * an async operation on a sub-component (e.g. a single card on a page),
 * than a regular `<CircularProgress />` should likely
 * be used to avoid unnecessary information from the label.
 *
 */
export const SpinnerWithLabel: FC<SpinnerWithLabelProps> = ({
  children,
  spinnerSize = SPINNER_SIZE,
  typeVariant = 'h5',
}) => {
  return (
    <Grid
      container
      direction="column"
      alignItems="center"
      justify="center"
      spacing={2}
    >
      <Grid item>
        <CircularProgress size={spinnerSize} />
      </Grid>
      <Grid item>
        <Typography variant={typeVariant}>{children}</Typography>
      </Grid>
    </Grid>
  );
};
