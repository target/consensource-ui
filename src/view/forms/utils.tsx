import React from 'react';
import { Grid, Typography, TextFieldProps } from '@material-ui/core';
import { BatchStatusRes } from 'services/api';

export interface TransactionFormProps {
  setBatchStatusLink: (statusLink: BatchStatusRes['link']) => void;
}

export type onChangeEvent = Parameters<
  NonNullable<TextFieldProps['onChange']>
>[0];

/**
 * Helper function that checks if any values in `state` have not been set.
 * Returns true if there is any key in `state` that is falsey.
 */
export function hasEmptyFields<T>(state: T) {
  return Object.keys(state).some((key) => !state[key as keyof T]);
}

export interface FormErrMsgProps {
  msg?: string;
}

export function FormErrMsg({ msg }: FormErrMsgProps) {
  if (!msg) {
    return null;
  }

  return (
    <Grid item>
      <Typography variant="caption" color="error">
        {msg}
      </Typography>
    </Grid>
  );
}
