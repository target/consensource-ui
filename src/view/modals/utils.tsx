import { DialogProps as MUIDialogProps } from '@material-ui/core';
import { FactoryResData } from 'services/api';

export interface DialogProps extends MUIDialogProps {
  handleClose: () => void;
}

export interface OrgDialogProps extends DialogProps {
  existing_org: FactoryResData;
}
