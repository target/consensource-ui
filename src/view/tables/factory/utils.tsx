import { MUIDataTableOptions, MUIDataTableChip } from 'mui-datatables';

export const DEFAULT_ROWS_PER_PAGE = 100;

export const textLabels: MUIDataTableOptions['textLabels'] = {
  body: {
    noMatch: 'No factories found',
  },
};

/**
 * Default props for the filter chips displayed in the
 * `<TableFilterList />`.
 */
export const filterChipProps: MUIDataTableChip = {
  color: 'secondary',
  variant: 'default',
};

/**
 * Utility function to convert a string to an array, or return
 * the original param if an array was passed.
 *
 * This operation is frequent when decoding URL query params.
 */
export const convertStrToArray = (val: string[] | string) => {
  return typeof val === 'string' ? [val] : val;
};
