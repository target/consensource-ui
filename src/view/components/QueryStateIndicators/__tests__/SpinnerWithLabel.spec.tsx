import React from 'react';
import { render } from 'utils/test-utils';
import { SpinnerWithLabel } from '../SpinnerWithLabel';

describe('<SpinnerWithLabel />', () => {
  it('renders', () => {
    const { container } = render(<SpinnerWithLabel>test</SpinnerWithLabel>);
    expect(container).toMatchSnapshot();
  });

  it('renders with a spinner size', () => {
    const { container } = render(
      <SpinnerWithLabel spinnerSize={100}>test</SpinnerWithLabel>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with a type variant', () => {
    const { container } = render(
      <SpinnerWithLabel typeVariant="h1">test</SpinnerWithLabel>,
    );
    expect(container).toMatchSnapshot();
  });
});
