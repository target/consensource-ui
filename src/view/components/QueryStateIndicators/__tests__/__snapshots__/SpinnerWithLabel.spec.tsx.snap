// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpinnerWithLabel /> renders 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 0px; height: 0px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        test
      </h5>
    </div>
  </div>
</div>
`;

exports[`<SpinnerWithLabel /> renders with a spinner size 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 100px; height: 100px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        test
      </h5>
    </div>
  </div>
</div>
`;

exports[`<SpinnerWithLabel /> renders with a type variant 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 0px; height: 0px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1"
      >
        test
      </h1>
    </div>
  </div>
</div>
`;
